R8 failure on release build >= 2.14alpha24 w/AGP 4.1.0

###### Reproduction Steps

1. check out tag 2.14alpha24
2. Bump android gradle plugin to 4.1.0
3. do an assembleRelease (or trigger minify some other way, but the `publishReleaseApk` target was where this was noticed)


###### Expected Result

Successful APK generation

###### Actual Result

NullPointerException during the R8 process

###### Workaround

I have temporarily reverted the set of changes for android gradle plugin 4.0.x -> 4.1.x update

- the AGP bump itself
- the Triple-T play publisher upgrade to 3.x
- the different way of setting play publisher gradle settings required by Triple-T 3.x


###### Research

There was nothing useful on the internet I could see, I haven't searched Google IssueTracker yet

This is still a problem with AGP4.1.1 (I just tried it today)

```
> Task :AnkiDroid:minifyReleaseWithR8 FAILED
R8: java.lang.NullPointerException

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':AnkiDroid:minifyReleaseWithR8'.
> com.android.tools.r8.CompilationFailedException: Compilation failed to complete

```

There were release notes indications in 4.1.1 that they had fixed something similar but it does not appear to have worked 
https://androidstudio.googleblog.com/2020/11/android-studio-411-available.html

I am pursuing this upstream https://issuetracker.google.com/issues/169584856#comment11

--- a/AnkiDroid/build.gradle
+++ b/AnkiDroid/build.gradle
@@ -238,7 +238,7 @@ dependencies {
     // #6419  - API 27 (& maybe others) could not perform new ZipFile() on a 2GB+ apkg
     // noinspection GradleDependency - pinned at 1.12 until API21 minSdkVersion (File.toPath usage)
     implementation 'org.apache.commons:commons-compress:1.12'
-    implementation 'androidx.constraintlayout:constraintlayout:2.0.3'
+    implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
 
 
     // May need a resolution strategy for support libs to our versions

--- a/AnkiDroid/build.gradle
+++ b/AnkiDroid/build.gradle
@@ -3,7 +3,7 @@ import com.android.build.OutputFile
 
 plugins {
  // Gradle plugin portal 
- id 'com.github.triplet.play' version '2.8.0'
+ id 'com.github.triplet.play' version '3.0.0'
 }
 
 apply plugin: 'com.android.application'

--- a/AnkiDroid/build.gradle
+++ b/AnkiDroid/build.gradle
@@ -179,8 +179,8 @@ android {
 }
 
 play {
-    serviceAccountCredentials = file("${homePath}/src/AnkiDroid-GCP-Publish-Credentials.json")
-    track = 'alpha'
+    serviceAccountCredentials.set(file("${homePath}/src/AnkiDroid-GCP-Publish-Credentials.json"))
+    track.set('alpha')
 }
 amazon {
     securityProfile = file("${homePath}/src/AnkiDroid-Amazon-Publish-Security-Profile.json")

--- a/build.gradle
+++ b/build.gradle
@@ -11,7 +11,7 @@ buildscript {
         }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.0.2'
+        classpath 'com.android.tools.build:gradle:4.1.1'
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
         classpath "app.brant:amazonappstorepublisher:0.1.0"

--- a/build.gradle
+++ b/build.gradle
@@ -9,9 +9,28 @@ buildscript {
         maven {
             url "https://plugins.gradle.org/m2/"
         }
+        // Temporary, needed while investigating https://github.com/ankidroid/Anki-Android/issues/7558
+        maven {
+            url 'https://storage.googleapis.com/r8-releases/raw'
+        }
+        maven {
+            url "https://storage.googleapis.com/r8-releases/raw/master"
+        }
+        // End temporary #7558 work
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.1.1'
+        // Temporary, for #7558 where AGP4.0.x works, 4.1.x fails, from https://issuetracker.google.com/issues/169584856
+        // Without workarounds we get 'com.android.tools.r8.CompilationFailedException: Compilation failed to complete'
+        // test case is to run `./gradlew assembleRelease` - if it gets to the point where it fails on keystore password, R8 already worked == success
+        //classpath 'com.android.tools:r8:25999632a8450072e0e39f2b28085ccb9766203f'  // This is diagnostic. Should Assertion error. Works?
+        //classpath 'com.android.tools:r8:89ac4c72113fdb73b43159522c932a1665de96c9' // This is diagnostic. Expect Assertion failure. Works?
+        //classpath 'com.android.tools:r8:2.2.28' // Release version, should work. Works.
+        //classpath 'com.android.tools:r8:2.1.74' // Cherry-picked / backport release, should work, fails with NullPointerException.
+        //classpath 'com.android.tools:r8:2.1.80' // Current stable release matched to AGP4.1.1 I think. Should work, fails with NullPointerException.
+        classpath 'com.android.tools:r8:2.2.40' // Current stable release matched to AGP4.2 I think. Should work.
+        // End of Temporary #7558 work
+
+        classpath 'com.android.tools.build:gradle:4.1.1' // This includes R8, and future versions will allow removal of #7558 workarounds.
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
         classpath "app.brant:amazonappstorepublisher:0.1.0"

--- a/lint-rules/build.gradle
+++ b/lint-rules/build.gradle
@@ -16,5 +16,5 @@ dependencies {
 
     testImplementation "junit:junit:4.13.1"
     testImplementation "com.android.tools.lint:lint:27.1.0"
-    testImplementation "com.android.tools.lint:lint-tests:27.1.0"
+    testImplementation "com.android.tools.lint:lint-tests:27.1.1"
 }

--- a/lint-rules/build.gradle
+++ b/lint-rules/build.gradle
@@ -12,7 +12,7 @@ repositories {
 
 dependencies {
     compileOnly "com.android.tools.lint:lint-api:27.1.1"
-    compileOnly "com.android.tools.lint:lint:27.1.0"
+    compileOnly "com.android.tools.lint:lint:27.1.1"
 
     testImplementation "junit:junit:4.13.1"
     testImplementation "com.android.tools.lint:lint:27.1.0"

--- a/lint-rules/build.gradle
+++ b/lint-rules/build.gradle
@@ -11,7 +11,7 @@ repositories {
 }
 
 dependencies {
-    compileOnly "com.android.tools.lint:lint-api:27.1.0"
+    compileOnly "com.android.tools.lint:lint-api:27.1.1"
     compileOnly "com.android.tools.lint:lint:27.1.0"
 
     testImplementation "junit:junit:4.13.1"

--- a/lint-rules/build.gradle
+++ b/lint-rules/build.gradle
@@ -15,6 +15,6 @@ dependencies {
     compileOnly "com.android.tools.lint:lint:27.1.1"
 
     testImplementation "junit:junit:4.13.1"
-    testImplementation "com.android.tools.lint:lint:27.1.0"
+    testImplementation "com.android.tools.lint:lint:27.1.1"
     testImplementation "com.android.tools.lint:lint-tests:27.1.1"
 }

